components/movementsensor/gpsnmea/gpsnmea.go

```diff
@@ -47,25 +47,27 @@ type I2CAttrConfig struct {
 }
 
 // Validate ensures all parts of the config are valid.
-func (cfg *AttrConfig) Validate(path string) error {
+func (cfg *AttrConfig) Validate(path string) ([]string, error) {
```

Comment:
(not for this pr) wdyt about centralizing i2c and serial validation logic somewhere? these validation blocks are going to be common.

Reply:
At some point yes. We can move the I2c config and Validation out of gpsnmea and use it everywhere. Problems are picking the most minimal config and validation checks because they can be very different based on component.

Reply:
Created a task.